integration-tests/source/devicedefender/DeviceDefenderIntegrationTests.cpp

```diff
@@ -0,0 +1,73 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "../IntegrationTestResourceHandler.h"
+#include <aws/core/Aws.h>
+#include <aws/iot/IoTClient.h>
+#include <gtest/gtest.h>
+
+using namespace Aws;
+using namespace Aws::Auth;
+using namespace Aws::Client;
+using namespace Aws::IoT;
+using namespace Aws::IoT::Model;
+using namespace std;
+
+extern std::string THING_NAME;
+extern std::string REGION;
+
+class TestDeviceDefenderFeature : public ::testing::Test
+{
+  public:
+    // cppcheck-suppress unusedFunction
+    void SetUp() override
+    {
+        SDKOptions options;
+        Aws::InitAPI(options);
+        {
+            ClientConfiguration clientConfig;
+            clientConfig.region = REGION;
+            resourceHandler =
+                unique_ptr<IntegrationTestResourceHandler>(new IntegrationTestResourceHandler(clientConfig));
+
+            resourceHandler->CreateAndAttachSecurityProfile(THING_NAME, metrics);
+        }
+    }
+    void TearDown() override
+    {
+        resourceHandler->DeleteSecurityProfile(THING_NAME);
+        SDKOptions options;
+        Aws::ShutdownAPI(options);
+    }
+    unique_ptr<IntegrationTestResourceHandler> resourceHandler;
+    vector<std::string> metrics{"aws:all-bytes-in", "aws:all-bytes-out", "aws:all-packets-in", "aws:all-packets-out"};
+};
+
+/**
+ * To test Device Defender we are creating a Security Profile to create violations for any metrics output and using this
+ * to facilitate testing this is a simple test to verify the metrics are being emitted by the Device Client by verifying
+ * all metrics are causing violations
```

Comment:
Just to be clear we are just trivially checking in bytes/packets in-out is less than 1 which should trigger a violation?

Reply:
Yes I can update that to make it clearer.

This was the only way I could think to test. DD has Device Side Metrics and Cloud Side Metrics and of course we only want to test the device and don't support custom metrics. So, verifying that we are in fact emitting these metrics seemed to be the best way.